homeassistant/components/assist_satellite/entity.py

```diff
@@ -0,0 +1,149 @@
+"""Assist satellite entity."""
+
+from abc import abstractmethod
+from collections.abc import AsyncIterable
+import time
+from typing import Final
+
+from homeassistant.components import stt
+from homeassistant.components.assist_pipeline import (
+    AudioSettings,
+    PipelineEvent,
+    PipelineEventType,
+    PipelineStage,
+    async_pipeline_from_audio_stream,
+)
+from homeassistant.const import EntityCategory
+from homeassistant.core import Context
+from homeassistant.helpers import entity
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.util import ulid
+
+from .models import AssistSatelliteEntityFeature, AssistSatelliteState, SatelliteConfig
+
+_CONVERSATION_TIMEOUT_SEC: Final = 5 * 60  # 5 minutes
+
+
+class AssistSatelliteEntity(entity.Entity):
+    """Entity encapsulating the state and functionality of an Assist satellite."""
+
+    entity_description = EntityDescription(
+        key="assist_satellite",
+        translation_key="assist_satellite",
+        entity_category=EntityCategory.CONFIG,
+    )
+    _attr_should_poll = False
+    _attr_state: AssistSatelliteState | None = None
+    _attr_supported_features: AssistSatelliteEntityFeature = (
+        AssistSatelliteEntityFeature(0)
+    )
+
+    _satellite_config = SatelliteConfig()
+
+    _conversation_id: str | None = None
+    _conversation_id_time: float | None = None
+
+    _run_has_tts: bool = False
+
+    async def _async_accept_pipeline_from_satellite(
+        self,
+        context: Context,
+        audio_stream: AsyncIterable[bytes],
+        start_stage: PipelineStage = PipelineStage.STT,
+        end_stage: PipelineStage = PipelineStage.TTS,
+        wake_word_phrase: str | None = None,
+    ) -> None:
+        """Triggers an Assist pipeline in Home Assistant from a satellite."""
+        device_id: str | None = None
+        if self.registry_entry is not None:
+            device_id = self.registry_entry.device_id
+
+        # Reset conversation id, if necessary
+        if (self._conversation_id_time is None) or (
+            (time.monotonic() - self._conversation_id_time) > _CONVERSATION_TIMEOUT_SEC
+        ):
+            self._conversation_id = None
+
+        if self._conversation_id is None:
+            self._conversation_id = ulid.ulid()
+
+        # Update timeout
+        self._conversation_id_time = time.monotonic()
+
+        # Set entity state based on pipeline events
+        self._run_has_tts = False
+        self._set_state(AssistSatelliteState.LISTENING)
```

Comment:
If start stage is wake word, it would be something else. Should we enforce only a couple of start stages to keep code simple today?

Reply:
I don't think we should because of the way I plan to implement remote pipeline triggering: the satellite will call back through this function. I just need to set the state more carefully with the pipeline events.